src/secret/mod.rs

```diff
@@ -217,13 +242,34 @@ impl FromStr for SecretManager {
             #[cfg(feature = "ledger")]
             SecretManagerDto::LedgerNanoSimulator => Self::LedgerNanoSimulator(LedgerSecretManager::new(true)),
 
-            SecretManagerDto::Mnemonic(mnemonic) => {
-                Self::Mnemonic(MnemonicSecretManager::try_from_mnemonic(&mnemonic)?)
-            }
+            SecretManagerDto::Mnemonic(mnemonic) => Self::Mnemonic(MnemonicSecretManager::try_from_mnemonic(mnemonic)?),
         })
     }
 }
 
+impl From<&SecretManager> for SecretManagerDto {
+    fn from(value: &SecretManager) -> Self {
+        match value {
+            #[cfg(feature = "stronghold")]
+            SecretManager::Stronghold(stronghold_dto) => Self::Stronghold(StrongholdDto {
+                password: None,
+                snapshot_path: stronghold_dto
+                    .snapshot_path
+                    .as_ref()
+                    .map(|s| s.clone().into_os_string().to_string_lossy().into()),
+            }),
+
+            #[cfg(feature = "ledger")]
+            SecretManager::LedgerNano(_) => Self::LedgerNano,
+
+            #[cfg(feature = "ledger")]
+            SecretManager::LedgerNanoSimulator(_) => Self::LedgerNanoSimulator,
+
+            SecretManager::Mnemonic(_mnemonic) => Self::Mnemonic("...".to_string()),
```

Comment:
`...` ?

Reply:
I'm not sure how to handle this best, the Seed in `MnemonicSecretManager(Seed)` doesn't has `Debug, Display` implemented and in the current use cases of the client/wallet we also don't need to convert it in this direction with the mnemonic/seed, we only need to know the type, because we don't want to store the actual seed/mnemonic somewhere